<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node pkg="sigverse_ros_bridge" exec="sigverse_ros_bridge" args="$(var sigverse_ros_bridge_port)" />
	
	<include file="$(find-pkg-share rosbridge_server)/launch/rosbridge_websocket_launch.xml" > 
		<arg name="port" value="$(var ros_bridge_port)"/>
		<arg name="default_call_service_timeout" value="5.0"/>
		<arg name="call_services_in_new_thread" value="true"/>
		<arg name="send_action_goals_in_new_thread" value="true"/>
	</include>

	<include file="$(find-pkg-share hsr_joint_state_2_joint_trajectory)/launch/display_launch.xml"/>

	<node pkg="hsr_joint_state_2_joint_trajectory" exec="hsr_joint_state_2_joint_trajectory">
		<remap from = "/joint_states" to = "/manual/joint_states"/>
		<!--
		<remap from = "/hsrb/arm_trajectory_controller/command"  to = "/hsrb/arm_trajectory_controller/command" />
		<remap from = "/hsrb/head_trajectory_controller/command" to = "/hsrb/head_trajectory_controller/command" />
		<remap from = "/hsrb/gripper_controller/command"         to = "/hsrb/gripper_controller/command" />
		-->
	</node>
</launch>

